shopify_integration/fulfilments.py

```diff
@@ -134,12 +133,28 @@ def create_delivery_notes(
 
 
 def update_fulfillment_items(
-	dn_items: List["DeliveryNoteItem"],
-	fulfillment_items: List["LineItem"]
+	so: List["SalesOrder"],
+	fulfillment_items: List["LineItem"],
 ):
-	for dn_item in dn_items:
+	dn_items= []
+	for so_item in so.items:
 		# TODO: figure out a better way to add items without setting valuation rate to zero
-		dn_item.allow_zero_valuation_rate = True
 		for item in fulfillment_items:
-			if get_item_code(item) == dn_item.item_code:
-				dn_item.qty = item.attributes.get("quantity")
+			if get_item_code(item) == so_item.item_code:
+				dn_item = frappe.new_doc("Delivery Note Item")
+				dn_item.update({
+					"allow_zero_valuation_rate": True, # is it necesarry now? #L151
+					"item_code": so_item.item_code,
+					"item_name": so_item.item_name,
+					"description": so_item.description,
+					"qty": item.attributes.get("quantity"),
+					"rate": so_item.rate,
+					"against_sales_order": so.name,
+					"shopify_order_item_id": item.attributes.get("id"),
+					"parenttype": "Delivery Note",
+					"parentfield": "items",
+				})
+				dn_items.append(dn_item)
```

Comment:
Can we pass the `dn` to this function, and use `dn.append("items", {...})` instead? Or do we need to create a child object only? That way, you'll have to set less fields than you're required to do.

Reply:
yeah makes sense! will try that and push